src/chocolatey/infrastructure.app/commands/ChocolateyApiKeyCommand.cs

```diff
@@ -53,28 +50,57 @@ public virtual void configure_argument_parser(OptionSet optionSet, ChocolateyCon
                 .Add("k=|key=|apikey=|api-key=",
                      "ApiKey - The API key for the source. This is the authentication that identifies you and allows you to push to a source. With some sources this is either a key or it could be a user name and password specified as 'user:password'.",
                      option => configuration.ApiKeyCommand.Key = option.remove_surrounding_quotes())
-                .Add("rem|remove",
-                    "Removes an API key from Chocolatey (DEPRECATED)",
-                    option => configuration.ApiKeyCommand.Remove = true)
                 ;
         }
 
         public virtual void handle_additional_argument_parsing(IList<string> unparsedArguments, ChocolateyConfiguration configuration)
         {
             configuration.Input = string.Join(" ", unparsedArguments);
+
+            if (unparsedArguments.Count > 1)
+            {
+                throw new ApplicationException("A single sources command must be listed. Please see the help menu for those commands");
+            }
+
+            var command = ApiKeyCommandType.Unknown;
+            string unparsedCommand = unparsedArguments.DefaultIfEmpty(string.Empty).FirstOrDefault();
+
+            if (!Enum.TryParse(unparsedCommand, true, out command) || command == ApiKeyCommandType.Unknown)
+            {
+                command = ApiKeyCommandType.List;
+
+                if (!string.IsNullOrWhiteSpace(unparsedCommand))
+                {
+                    this.Log().Warn("Unknown command {0}. Setting to list.".format_with(unparsedCommand));
+                }
+                else if (!string.IsNullOrWhiteSpace(configuration.ApiKeyCommand.Key))
+                {
+                    this.Log().Warn("Apikey provided. Setting command to add.");

```

Comment:
Is this the right casing?  Seems like we have lots of different ways of writing this :-(

`ApiKey`, `API Key`, `API key`, `Apikey`

Reply:
Yeah I wasn't sure, what's our favourite way to write it? 😂

Reply:
Based on the current usage, i.e `Added ApiKey` and `Updated ApiKey`, let's go with `ApiKey`

Reply:
Sorted :)